library/src/main/java/com/fidesmo/fdsm/FidesmoApiClient.java

```diff
@@ -241,15 +242,23 @@ public static String getVersion() {
 
     // Prefer English if system locale is not present
     // to convert a possible multilanguage node to a string
-    public static String lamei18n(JsonNode n) {
+        public static String lamei18n(JsonNode n) {
         // For missing values
         if (n == null)
             return "";
-        if (n.size() > 0) {
-            Map<String, Object> langs = mapper.convertValue(n, new TypeReference<Map<String, Object>>() {
-            });
-            Map.Entry<String, Object> first = langs.entrySet().iterator().next();
-            return langs.getOrDefault(Locale.getDefault().getLanguage(), langs.getOrDefault("en", first.getValue())).toString();
+        if (!n.isEmpty()) {
+            //Check if JSON comes in new Translation format, it can be multilanguage
+            JsonNode jsonNodeFormat = (n.has("text")) ? n.get("text") : n;
+            try {
+                Map<String, Object> langs = mapper.convertValue(jsonNodeFormat, new TypeReference<Map<String, Object>>() {
```

Comment:
the text node will always be a single string for the client, But it might also contain parameters so we need to susbtitute them in string.